pkg/workloadmeta/store.go

```diff
@@ -36,48 +36,12 @@ type subscriber struct {
 	filter *Filter
 }
 
-type sourceToEntity map[Source]Entity
-
-func (s sourceToEntity) merge(sources []Source) Entity {
-	if len(sources) == 0 {
-		sources = s.sources()
-	}
-
-	var merged Entity
-	for _, source := range sources {
-		if e, ok := s[source]; ok {
-			if merged == nil {
-				merged = e.DeepCopy()
-			} else {
-				err := merged.Merge(e)
-				if err != nil {
-					log.Errorf("cannot merge %+v into %+v: %s", merged, e, err)
-				}
-			}
-		}
-	}
-
-	return merged
-}
-
-func (s sourceToEntity) sources() []Source {
-	sources := make([]Source, 0, len(s))
-
-	for source := range s {
-		sources = append(sources, source)
-	}
-
-	sort.SliceStable(sources, func(i, j int) bool { return sources[i] < sources[j] })
-
-	return sources
-}
-
 // store is a central storage of metadata about workloads. A workload is any
 // unit of work being done by a piece of software, like a process, a container,
 // a kubernetes pod, or a task in any cloud provider.
 type store struct {
 	storeMut sync.RWMutex
-	store    map[Kind]map[string]sourceToEntity
+	store    map[Kind]map[string]*cachedEntity
```

Comment:
I think for readability, it would be good to clarify somehow (comment, type alias, etc.) that the string in the second map is an Entity ID.